Nate Pudwell | x86_toolchain.sh addons.

Ohtheirony06 opened the pull request:
**Changes**

1. BIT variable is True by default.

2. An option to turn BIT variable off.

3. Overwrite protection -Program will check at the start if the files that are going to be created already exist or not. If they do and the -ow is not used the program will not allow you to overwrite and exit.

4. Copy option - If -ow is enabled and it finds a that it will overwrite the file. If this is the case x86_toolchain.sh will ask the user for a name they want to copy the object file to. (executable will not be copied) If the filename of the copy already exists it will exit the program after telling the user that. 

5. Linking option used by gcc (-l). If the user wants to use gcc for linking the object file they can use -l. otherwise if -l is not used it will use the normal method.
GCC used for this:   gcc -no-pie -o $OUTPUT_FILE $OUTPUT_FILE.o -nostartfiles -e _start
If it doesn't detect _start it will default to 0000000000401020


6. Execute option (-e). If this option is enabled x86_toolchain.sh will attempt to execute the file in the local directory.

7. Show shared library option (-ldd). If the user wants to see the shared libraries used by the executable they can use -ldd.

8. Option for 32 bits if they want to turn off BITs. (-32)

9. Added multiple checks along the line with colored output to tell the user if something was created properly ornot created properly.

10. The program uses gcc and the installer does not check if gcc is installed. Added a gcc check into x86_toolchain.sh if user is using -l option it will check if gcc is installed if not it will print that it's not installed and exit. If it is installed it will print the that it's installed and the gcc version the user is using.

11. The new options implemented into x86_toolchain.sh have been printed in blue.

NOTE: No functionality of x86_toolchain.sh has been removed and all default options still work. Only thing to keep in mind is that overwrite protection is on and it's x86-64 default.

onisoyyc commented:
First Branch "gcc-version" had some serious issues setting up the pull request, so will work on this one instead.

LubosKuzma commented:
Thanks, this is a great work

Ohtheirony06 committed:
Nate Pudwell | x86_toolchain.sh addons.

Changes
BIT variable is True by defaulting.
An option to turn BIT variable off.

Overwrite protection -Program will check at the start if the files that are going to be created already exist or not. If they do and the -ow is not used the program will not allow you to overwrite and exit.

Copy option - If -ow is enabled and it finds a that it will overwrite the file. If this is the case x86_toolchain.sh will ask the user for a name they want to copy the object file to. (executable will not be copied) If the filename of the copy already exists it will exit the program after telling the user that. 

Linking option used by gcc (-l). If the user wants to use gcc for linking the object file they can use -l. otherwise if -l is not used it will use the normal method.
GCC used for this:   gcc -no-pie -o $OUTPUT_FILE $OUTPUT_FILE.o -nostartfiles -e _start
If it doesn't detect _start it will default to 0000000000401020


Execute option (-e). If this option is enabled x86_toolchain.sh will attempt to execute the file in the local directory.

Show shared library option (-ldd). If the user wants to see the shared libraries used by the executable they can use -ldd.

Option for 32 bits if they want to turn off BITs. (-32)

Added multiple checks along the line with colored output to tell the user if something was created properly ornot created properly.

The program uses gcc and the installer does not check if gcc is installed. Added a gcc check into x86_toolchain.sh if user is using -l option it will check if gcc is installed if not it will print that it's not installed and exit. If it is installed it will print the that it's installed and the gcc version the user is using.

The new options implemented into x86_toolchain.sh have been printed in blue.

LubosKuzma commented:
Amazing job. Thank you!
